lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -2746,6 +2745,129 @@ LogicalResult ConvertAtenOp<AtenTransposeIntOp>::matchAndRewrite(
   return success();
 }
 
+template <>
+LogicalResult ConvertAtenOp<AtenMaxDimOp>::matchAndRewrite(
+    AtenMaxDimOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+
+  auto selfType = adaptor.self().getType().dyn_cast<TensorType>();
+  if (!selfType)
+    return op.emitError("Only tensor types are supported");
```

Comment:
Use rewriter.notifyMatchFailure inside patterns to document limitations of patterns instead.